conformance/service_import.go

```diff
@@ -83,4 +85,71 @@ var _ = Describe("", func() {
 			Expect(net.ParseIP(serviceImport.Spec.IPs[0])).ToNot(BeNil(),
 				reportNonConformant(fmt.Sprintf("The value %q is not a valid IP", serviceImport.Spec.IPs[0])))
 		})
+
+	Context("A service exported on two clusters", func() {
+		var helloService2 *corev1.Service
+
+		BeforeEach(func() {
+			requireTwoClusters()
+
+			helloService2 = newHelloService()
+		})
+
+		JustBeforeEach(func() {
+			// Sleep a little before deploying on the second cluster to ensure the first cluster's ServiceExport timestamp
+			// is older so conflict checking is deterministic.
+			time.Sleep(100 * time.Millisecond)
+
+			t.deployHelloService(&clients[1], helloService2)
+			t.createServiceExport(&clients[1])
+		})
+
+		Context("", func() {
+			BeforeEach(func() {
+				helloService2.Spec.Ports = []corev1.ServicePort{
+					t.helloService.Spec.Ports[0],
+					{
+						Name:     "stcp",
+						Port:     142,
+						Protocol: corev1.ProtocolSCTP,
+					},
+				}
+			})
+
+			Specify("should expose the union of the constituent service ports", Label(RequiredLabel), func() {
+				AddReportEntry(SpecRefReportEntry, "https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/1645-multi-cluster-services-api#service-port")
+
+				serviceImport := t.awaitServiceImport(&clients[0], t.helloService.Name, func(serviceImport *v1alpha1.ServiceImport) bool {
+					return len(serviceImport.Spec.Ports) == 3
+				})
+				Expect(serviceImport).NotTo(BeNil(), "ServiceImport was not found")
```

Comment:
```suggestion
				Expect(serviceImport).NotTo(BeNil(), "ServiceImport was not found or did not have the expected number of ports")
```

Reply:
`awaitServiceImport` only returns nil if the `ServiceImport` is not found. If the verification function isn't satisfied then `awaitServiceImport` times out but still succeeds and returns the `ServiceImport`, enabling more granular and exact assertions by the caller with richer descriptions and avoiding vaguer description as suggested above.